### DISCUSSION TITLE
Pengurusan Khairat (Enhancement)

### DISCUSSION THREAD

**Discussion by rohaizanr** [Dev4w4n/e-masjid.my#143] [Ideas]
Skrin khairat yang sedia ada

Bahagian Daftar:
<img width="1034" alt="Screenshot 2024-03-16 at 1 30 39 AM" src="https://github.com/Dev4w4n/e-masjid.my/assets/11247162/4d3d6bc2-3916-4981-bcf0-c8538730b4db">

Bahagian Carian:
<img width="1026" alt="Screenshot 2024-03-16 at 1 31 23 AM" src="https://github.com/Dev4w4n/e-masjid.my/assets/11247162/7665b38b-ef15-4b5e-a120-c110a80f0bc5">

Keperluan terkini:

1. Tiada paparan "Payment History"
2. Tiada cara untuk masukkan bayaran lebih dari setahun
3. Tiada cara untuk masukkan bayaran tahun tertunggak
4. Tiada cara untuk menanda ahli sudah mati (Jika ahli sudah mati, tanggungan boleh dipindah ke ahli khairat yang baru)

**Comment by DeenSkuad** [comment]
1. Paparan "Payment History" clear by tahun
![Khariat -Carian -Pilihan 2 -Status Bayaran Tahunan](https://github.com/Dev4w4n/e-masjid.my/assets/157102745/ae2fcc57-c1bd-4543-b18a-fc7338e4b5b7)

2. Bayaran Tunggakan Bulanan / Tahunan Tunai Modal
- Boleh abaikan tunai or online jika tidak perlu

![Tunai -Status Bayaran Tahunan](https://github.com/Dev4w4n/e-masjid.my/assets/157102745/7abbfca1-05ca-4b83-b8ca-7c465123a567)

3. Bayaran Tunggakan Bulanan / Tahunan Online Modal
![Khariat -Carian -Pilihan 2 -Status Bayaran Tahunan-1](https://github.com/Dev4w4n/e-masjid.my/assets/157102745/358ad003-0c07-4828-b1cc-8c654a66c44f)

4. Status Hidup
![Khariat -Carian-A](https://github.com/Dev4w4n/e-masjid.my/assets/157102745/bd1b7e9b-9382-42e3-855e-bf0a0d6cfeba)

Jika tidak keberatan boleh comment di figma
https://www.figma.com/file/YJV6fF5AnI7YxbhUWfDNqU/e-masjid?type=design&node-id=0%3A26&mode=design&t=v8Ss3dAT83zc8PfN-1

**Comment by mfar1984** [comment]
Penanda tu kalau tukar ke nama Mukim boleh tak? Tapi penanda tu, kalau dapat, kita masukkan bahagian Setting > General > Bawah nama Kawasan > Mukim

Tujuan : supaya e masjid lain boleh guna nama berlainan mukim

Get file dari Setting > General > Mukim Tu ke dalam form Daftar Ahli

**Comment by mfar1984** [comment]
Untuk pendaftaran Ahli, sekiranya IC bertindih, maksudnya ahli tersebut sudah terdaftar sebagai ahli khariah atau id (ic) tersebut telah dimasukkan di dalam daftar kubur atau daftar jenazah

**Comment by DeenSkuad** [comment]
4. Carian > Tambahan Wakil
ask = perlukah tindakan dalam table ? jika tidak boleh diremove
![Khariat -Carian-A](https://github.com/Dev4w4n/e-masjid.my/assets/157102745/e313dd23-b1a2-4e79-8b54-b2793b60e5fe)

**Comment by DeenSkuad** [comment]
1. Tiada paparan "Payment History" 
digambarkan action seterusnya adalah:
Table > Tindakan > Bayaran modal


ask = action seterusnya ? saya agak blur disini

exp = table > Modal bayaran  > hantar notis / atau tunai ?


![Khariat -Carian -Pilihan 2 -Status Bayaran Tahunan](https://github.com/Dev4w4n/e-masjid.my/assets/157102745/5a1cd323-651f-48c2-926a-453dc27f7147)

![Tunai -Status Bayaran Tahunan](https://github.com/Dev4w4n/e-masjid.my/assets/157102745/147421e7-a6c2-4595-a208-faafe6196db1)

**Comment by dinayub** [comment]
Carian Table
![Khariat -Carian-A](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/d2ae04a1-7810-4291-a2da-e214202673eb)

**Reply by dinayub** [reply]
Carian
- Remove Wakil.  dah ada penanda
- Remove Tindakan - tidak perlu
![Khariat -Carian-A](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/df19cf19-b7f4-439b-ba6a-e72f3703f839)

**Reply by dinayub** [reply]
Pindaan Ahli Keluarga 
- Tambah Status Hidup
- Tindakan 
   - Lihat = user/id, 
   - Ketua Keluarga
   
Mungkin Lebih Senang Auto jadi Ahli Keluarga
![Khariat -Carian-A - Pindaan Ahli Keluarga](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/cfc0caf9-77e1-4dd0-be4c-36fa0b47e164)

**Comment by dinayub** [comment]
Penyata Bayaran Table
![Khariat -Carian -Pilihan 2 -Status Bayaran Tahunan](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/32bfb1c0-8ea0-4fe8-9723-8b66b6587dfa)

Tindakan > Terima / Hantar Notis Tunggakkan
![Tunai -Status Bayaran Tahunan](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/1531786c-3817-4de2-9c5a-71461556b826)

**Reply by dinayub** [reply]
Penyata Bayaran Table
-  Remove Tunggakan
- Tambah tindakan
![Khariat -Carian -Table Status Bayaran Tahunan](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/9d47e509-c38f-4201-b6ed-1700f4ad6b85)

![Khariat -Carian -Status Bayaran Tahunan -modal](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/b4869993-2704-448c-b434-3fe207598920)
![Khariat -Carian -Status Bayaran Tahunan -modal-1](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/6b0a3200-6acc-4cd2-b87e-5e35b2a4cc25)

**Reply by rohaizanr** [reply]
terima kasih @dinayub 

nak tanya, boleh tak bagi nampak sikit hubungan data main screen "Status Bayaran Tahunan" dengan side panel tu. saya tak berapa nampak relationship dia.

![Screenshot 2024-03-23 at 2 24 59 AM](https://github.com/Dev4w4n/e-masjid.my/assets/11247162/a315ec82-6cb6-4dc5-aa10-f98c0d292648)

![Screenshot 2024-03-23 at 2 27 49 AM](https://github.com/Dev4w4n/e-masjid.my/assets/11247162/7d58de49-fcb2-4d7d-88ab-869d861bced3)

**Reply by DeenSkuad** [reply]
Xpa nti saya adjust. Maybe xperlu modal. Terus ada tab bayaran Dan tunggakkan

**Reply by dinayub** [reply]
Saya rasa ini lebih mudah faham
-  Filter : Cetak, Bayaran, Tunggakkan
-  Tindakan: Lihat, Cetak, Hantar Notis

![Khariat -Carian -Table Status Bayaran Tahunan](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/5a0f6b75-cc7d-4a0a-95a0-18e711572387)

**Reply by rohaizanr** [reply]
ok macam ni nampak clear sikit.

button tindakan akan buat apa?

**Reply by dinayub** [reply]
button tindakan : Lihat Resit, Terima Bayaran

**Comment by mfar1984** [comment]
Boleh tahu tak pembayaran tu memang untuk setahun terus atau ada option untuk bulanan?

**Reply by rohaizanr** [reply]
selalunya skim khairat is bayaran tahunan.

tak pernah jumpa lagi bayaran bulanan, kalau ada mcm merugikan ahli khairat. 

kos pengkebumian dalam RM1200.

**Reply by imie** [reply]
biasa khairat bayar setahun terus sebab tak mahal. tempat saya bayar setahun untuk satu family rm20 je.

**Reply by imie** [reply]
![image](https://github.com/Dev4w4n/e-masjid.my/assets/2321338/8ac65f94-0b8f-47e4-9b24-48f4274977e7)

**Reply by rohaizanr** [reply]
@dinayub kalau mcm ni, kena ada textbox untuk key-in value. senang

**Reply by dinayub** [reply]
Mungkin Boleh dibuat mcm ni
- Pilih Tahun
- Text RM
![Khariat -Carian -Status Bayaran Tahunan Tindakan-modal](https://github.com/Dev4w4n/e-masjid.my/assets/164418432/34465276-35e0-4757-8b3f-33134d06f079)

**Reply by rohaizanr** [reply]
nice @dinayub 

boleh wrap up dah rasanya

**Reply by dinayub** [reply]
kira boleh ui code la ni ?. lain2 ada nk tambah ?

**Reply by rohaizanr** [reply]
opps @dinayub tertinggal yang ni

4. Tiada cara untuk menanda ahli sudah mati (Jika ahli sudah mati, tanggungan boleh dipindah ke ahli khairat yang baru)

**Reply by DeenSkuad** [reply]
di komen atas bang haji

**Reply by dinayub** [reply]
di komen atas bang haji

**Comment by lurhays** [comment]
Ni apa aku paham dari post2 kat atas so far..

[A] Scenario kematian ahli khairat:
1) Bila dapat notis kematian, AJK search ahli kat Khairat>Carian.
2) AJK click row ahli mati.
3) System display details ahli mati kat Khairat>Daftar.
4) AJK update form Daftar ahli/Status: Meninggal Dunia.
5) AJK update form Maklumat tanggungan/Tindakan: Ketua Keluarga

Soalan:
- Dalam senario ahli khairat mati, masa mana keluarga dia claim khairat? 
- Kalau claim khairat, perlu update kah untuk record sudah claim, masa, siapa terima dan kat flow mana? (Ada keperluan simpan transaction khairat secara keseluruhan kah?)
- Kalau ahli khairat yang mati tu ada tunggakan, apa jadi?
- Kalau ahli khairat yang mati tu takda family, takda tanggungan, masjid handle macam mana?


[B] Scenario pembayaran tahunan khairat:
1) AJK search ahli kat Khairat>Carian.
2) AJK click row ahli mati.
3) System display details ahli mati kat Khairat>Daftar.
<img width="440" alt="image" src="https://github.com/Dev4w4n/e-masjid.my/assets/5729687/7ccf5653-68dd-4dbc-ace6-b45aa2c55d7e">

4) AJK check checkbox row tahun mana nak bayar.
5) System update number of tahun dan Jumlah bayaran kat footer.
6) AJK click button "Terima Bayaran".

Soalan:
- Implementation sebelum ni capture resit number, in this case, kena capture resit number untuk setiap tahun or lumpsum resit number or tak perlu capture?
- Camna system tau berapa banyak ahli kena bayar untuk tahun tu? 
- Camna system tau nak kira hutang start dari tahun bila?


Untuk scenario ahli khairat overdue/hutang, hantar notis kat ahli by year atau summary? 
Kena record setiap kali AJK trigger hantar notis? (in case, hg tak habaq aku vs aku habaq dah 🤣)
Likely, perlu add 1 field untuk capture email address (and verify email validity somewhere kat public? modul anak kariah tu ka)

**Comment by rohaizanr** [comment]
@lurhays , terima kasih ..

Dalam senario ahli khairat mati, masa mana keluarga dia claim khairat?
**Pada hari kematian**

Kalau claim khairat, perlu update kah untuk record sudah claim, masa, siapa terima dan kat flow mana? (Ada keperluan simpan transaction khairat secara keseluruhan kah?)
**Sepatutnya ada boucher menunjukkan wang keluar. (tapi yang ni nanti dulu lah)**

Kalau ahli khairat yang mati tu ada tunggakan, apa jadi?
**Tak layak terima pada hari kematian**

Kalau ahli khairat yang mati tu takda family, takda tanggungan, masjid handle macam mana?
**Tak pernah lagi jumpa kes macam ni, tapi assume keluarga terdekat akan dihubungi lah**

Implementation sebelum ni capture resit number, in this case, kena capture resit number untuk setiap tahun or lumpsum resit number or tak perlu capture?
**ya perlu, @dinayub kita tertinggal yang ni**

Camna system tau berapa banyak ahli kena bayar untuk tahun tu?
**Semua masjid skim tak sama, jadi kita letak textbox manual key-in**

Camna system tau nak kira hutang start dari tahun bila?
**BIasanya sekarang ni diorg dah ada manual sistem dan ada rekod tu semua. Jadi, manually define**

Untuk scenario ahli khairat overdue/hutang, hantar notis kat ahli by year atau summary?
**Kalau nak advance, user akan notified thru Whatsapp, for a start tak bagi notis, tapi nanti user boleh search by IC thru public-web**

Kena record setiap kali AJK trigger hantar notis?
**Kalau ada notis, ya kena record, tapi not planned yet.**

**Comment by dinayub** [comment]
Implementation sebelum ni capture resit number, in this case, kena capture resit number untuk setiap tahun or lumpsum resit number or tak perlu capture?
ya perlu, @dinayub kita tertinggal yang ni

boleh imagine : 
- Table > Filter > input carian no resit
- Table > Filter > Senarai Bayaran / Bayaran, Tunggakkan
- Table > Calendar > select by Tahun
- Table > tindakan > modal resit

**Reply by rohaizanr** [reply]
Rasanya kita buat mcm ni, di Tetapan>Khairat, ada option 
1. auto gen resit no
2. Manual key in resit

Boleh tak @dinayub

**Reply by DeenSkuad** [reply]
Alamak xprasan 4hari dah. Boleh. Kira nk tgk ui setting kan?
